internal/core/build_info.go

```diff
@@ -56,48 +56,55 @@ func (b *BuildInfo) GetUserAgent() string {
 	return userAgentPrefix
 }
 
-func (b *BuildInfo) checkVersion() {
-	if !b.IsRelease() || os.Getenv(scwDisableCheckVersionEnv) == "true" {
-		logger.Debugf("skipping check version")
+func (b *BuildInfo) checkVersion(ctx context.Context) {
+	if !b.IsRelease() || ExtractEnv(ctx, scwDisableCheckVersionEnv) == "true" {
+		ExtractLogger(ctx).Debug("skipping check version")
 		return
 	}
 
-	latestVersionUpdateFilePath := getLatestVersionUpdateFilePath()
+	latestVersionUpdateFilePath := getLatestVersionUpdateFilePath(ExtractCacheDir(ctx))
 
 	// do nothing if last refresh at during the last 24h
 	if wasFileModifiedLast24h(latestVersionUpdateFilePath) {
-		logger.Debugf("version was already checked during past 24 hours")
+		ExtractLogger(ctx).Debug("version was already checked during past 24 hours")
 		return
 	}
 
 	// do nothing if we cannot create the file
-	if !createAndCloseFile(latestVersionUpdateFilePath) {
+	err := createAndCloseFile(latestVersionUpdateFilePath)
+	if err != nil {
+		ExtractLogger(ctx).Debug(err.Error())
 		return
 	}
 
 	// pull latest version
-	latestVersion, err := getLatestVersion()
+	latestVersion, err := getLatestVersion(ExtractHttpClient(ctx))
 	if err != nil {
-		logger.Debugf("failed to retrieve latest version: %s", err)
+		ExtractLogger(ctx).Debugf("failed to retrieve latest version: %s\n", err)
 		return
 	}
 
 	if b.Version.LessThan(latestVersion) {
-		logger.Warningf("a new version of scw is available (%s), beware that you are currently running %v", latestVersion, b.Version)
+		ExtractLogger(ctx).Warningf("a new version of scw is available (%s), beware that you are currently running %s\n", latestVersion, b.Version)
 	} else {
-		logger.Debugf("version is up to date (%s)", b.Version)
+		ExtractLogger(ctx).Debugf("version is up to date (%s)\n", b.Version)
 	}
 }
 
-func getLatestVersionUpdateFilePath() string {
-	return filepath.Join(scw.GetCacheDirectory(), latestVersionUpdateFileLocalName)
+func getLatestVersionUpdateFilePath(cacheDir string) string {
+	return filepath.Join(cacheDir, latestVersionUpdateFileLocalName)
 }
 
 // getLatestVersion attempt to read the latest version of the remote file at latestVersionFileURL.
-func getLatestVersion() (*version.Version, error) {
-	resp, err := (&http.Client{
-		Timeout: latestVersionRequestTimeout,
-	}).Get(latestVersionFileURL)
+func getLatestVersion(client *http.Client) (*version.Version, error) {
+	ctx, cncl := context.WithTimeout(context.Background(), latestVersionRequestTimeout)
+	defer cncl()
```

Comment:
Could we have a more explicit function name? what does cncl means?

Reply:
cancel